src/citrine/waiting/waiting.py

```diff
@@ -0,0 +1,77 @@
+from time import time, sleep
+from typing import Callable
+from pprint import pprint
+from citrine._rest.collection import Collection
+from citrine.resources.workflow_executions import (
+    WorkflowExecution,
+    WorkflowExecutionStatus,
+)
+
+
+def _wait_until(condition: Callable[[], bool], timeout: float = 900.0, interval: float = 3.0):
```

Comment:
The code would be more straightforward and readable if this `_wait_until` function was inline in `wait_until_validated` and `wait_until_finished`, rather than abstracted. Given that this function is only used twice, and requires the definition of separate functions that are passed around, you don't seems to be saving any lines of code by this abstraction.

Reply:
I agree that inlining probably makes more sense.  In particular, it should allow for better error messaging, because the logical content of the condition can be used to populate the error message.